src/parse.zig

```diff
@@ -571,12 +586,27 @@ const Parser = struct {
                             break :parse;
                         },
                         .NewLine => return error.UnexpectedToken,
-                        else => {},
+                        else => {
+                            var i: usize = tok.start;
+                            while (i < tok.end) : (i += 1) {
+                                var c = self.tree.source[i];
+                                switch (c) {
+                                    '\'' => {
+                                        try node.string_value.append(c);
```

Comment:
If we are going to store the literal value in an array list for this case, I think we ought to that for all cases to streamline the processing in `Yaml.Value`. This way, we won't have to special-case anything, simply always return `Value{ .string = self.string.value.items() }`.